src/reflect/scala/reflect/api/Annotations.scala

```diff
@@ -91,6 +91,20 @@ trait Annotations { self: Universe =>
     @deprecated("use `tree.children.tail` instead", "2.11.0")
     def scalaArgs: List[Tree]
 
+    /** For arguments in [[scalaArgs]], this method returns `true` if the argument AST is a default inserted
+     *  by the compiler, not an explicit argument passed in source code.
+     *
+     *  Since Scala 2.13.17, the defaults are ASTs of the default expression in the annotation definition.
+     *  Example:
+     *  {{{
+     *    class ann(x: Int = 42) extends Annotation
+     *    @ann class C
+     *  }}}
+     *  The `annotation.scalaArgs.head` is an AST `Literal(Constant(42))` for which the `argIsDefault` method
+     *  returns `true`.
+     */
+    def argIsDefault(tree: Tree): Boolean = true // default implementation to ensure backwards bin compat
```

Comment:
Isn't that implementation a lie? If it ends up being used, it will return the wrong result as often as the right one.

Shouldn't it try to implement the same test that would be done in user-space for previous compiler versions? That way, we could reliably depend on that method, and it would work on older compilers as well. Couldn't we?

Reply:
I was lazy I guess.. The only implementation (or now, override) of this method is in `reflect.internal.AnnotationInfo`, which is in the same jar (scala-reflect). So the default implementation here should never be used.

But I'll take a look if the actual implementation can be cleanly done in the trait here.

Reply:
Ah I did not notice it was in the same jar.

In that case, an alternative is to filter it out in the MiMa filter, with the justification that the trait is "effectively sealed" or something like that.

Reply:
Right, that's better.